api/v1alpha3/azuremachine_types.go

```diff
@@ -72,6 +72,18 @@ type AzureMachineSpec struct {
 	// AllocatePublicIP allows the ability to create dynamic public ips for machines where this value is true.
 	// +optional
 	AllocatePublicIP bool `json:"allocatePublicIP,omitempty"`
+
+	// SpotVMOptions allows the ability to specify the Machine should use a Spot VM
+	// +optional
+	SpotVMOptions *SpotVMOptions `json:"spotVMOptions,omitempty"`
+}
+
+// SpotVMOptions defines the options relevant to running the Machine on Spot VMs
+type SpotVMOptions struct {
+	// MaxPrice defines the maximum price the user is willing to pay for Spot VM instances
+	// +optional
+	// +kubebuilder:validation:pattern="^[0-9]+(\.[0-9]+)?$"
```

Comment:
I wonder if // +kubebuilder:validation:Type=number would be equivalent?

Reply:
I added this and it works! Means that the `maxPrice` must be specified without quotes now, so I've updated the doc too, is that preferred? Seems a shame we can't have both

Reply:
I think without quotes makes more sense anyways since to the user it's a number, not a string